src/containers/ImageUploader/CreateImage.tsx

```diff
@@ -12,35 +12,46 @@ import ImageForm from './components/ImageForm';
 import { createFormData } from '../../util/formDataHelper';
 import * as imageApi from '../../modules/image/imageApi';
 import { toEditImage } from '../../util/routeHelpers';
-import { NewImageMetadata } from '../../modules/image/imageApiInterfaces';
+import { ImageApiType, NewImageMetadata } from '../../modules/image/imageApiInterfaces';
 import { useLicenses } from '../Licenses/LicensesProvider';
 
 interface Props extends RouteComponentProps {
   isNewlyCreated?: boolean;
-  showSaved?: boolean;
+  editingArticle?: boolean;
+  onImageCreated?: (image: ImageApiType) => void;
+  closeModal?: () => void;
+  inModal?: boolean;
 }
 
-const CreateImage = ({ history, isNewlyCreated, showSaved }: Props) => {
+const CreateImage = ({
+  history,
+  isNewlyCreated,
+  editingArticle,
+  onImageCreated,
+  inModal,
+  closeModal,
+}: Props) => {
   const { i18n } = useTranslation();
   const locale = i18n.language;
   const { imageLicenses } = useLicenses();
 
   const onCreateImage = async (imageMetadata: NewImageMetadata, image: string | Blob) => {
     const formData = await createFormData(image, imageMetadata);
     const createdImage = await imageApi.postImage(formData);
-    if (!imageMetadata.id) {
+    onImageCreated?.(createdImage);
```

Comment:
Den var ny `onImageCreated?.(createdImage);`, fancy greier

Reply:
Det er berre en sjekk om funksjonen finnes før den brukes. Vi har det fleire plasser, også for arrays: `array?.[0]`.